mobile/src/src/components/Menu/Profile/EditComponent.jsx

```diff
@@ -0,0 +1,720 @@
+  ScrollView,
+} from 'react-native';
+import { Avatar, Button, TextInput } from 'react-native-paper';
+import { Picker } from '@react-native-picker/picker';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { Choices, ProfileEndpoint } from '../../../utils/endpoints';
+import Header from '../../common/Header';
+import ProfileComponents from './ProfileComponents';
+import * as ImagePicker from 'expo-image-picker';
+
+const EditComponent = ({ user, navigation }) => {
+  // useStates for data
+  const [bgPic, setbgPic] = useState();
+  const [pfp, setPfp] = useState();
+  //const [pfpFile, setPfpFile] = useState();
+  //const [bgPicFile, setbgPicFile] = useState();
+  const [prefName, setPrefName] = useState(user.preferred_name);
+  const [phone, setPhone] = useState(user.phone_number);
+  const [pronoun, setPronoun] = useState(user.pronouns);
+  const [birth, setBirth] = useState(user.birthday);
+  const [reg, setRegion] = useState(user.community);
+  const [program, setProgram] = useState(user.enrolled_program);
+  const [term, setTerm] = useState(user.current_term);
+
+  // useStates for modal
+  const [visiblePronoun, setVisiblePronoun] = useState(false);
+  const [visibleReg, setVisibleReg] = useState(false);
+  const [visiblePro, setVisiblePro] = useState(false);
+  const [visibleTerm, setVisibleTerm] = useState(false);
+  const [visibleBirth, setVisibleBirth] = useState(false);
+  const [visiblePics, setVisiblePics] = useState(false);
+
+  // useState for datetimepicker
+  const [date, setDate] = useState(new Date());
+
+  // useStates for stupid react native picker defaulting 0
+  const [zeroPronoun, setZeroPronoun] = useState();
+  const [zeroReg, setZeroReg] = useState();
+  const [zeroPro, setZeroPro] = useState();
+  const [zeroTerm, setZeroTerm] = useState();
+
+  const handleEdit = async () => {
+    const result = await ProfileEndpoint.patch({
+      //profile_photo: pfp ? pfpFile : user.profile_photo,
+      //background_photo: bgPic ? bgPicFile : user.background_photo,
+      preferred_name: prefName ? prefName : user.preferred_name,
+      phone_number: phone ? phone : user.phone_number,
+      pronouns: zeroPronoun == 0 ? 0 : pronoun ? pronoun : user.pronouns,
+      birthday: birth ? birth : user.birthday,
+      community: zeroReg == 0 ? 0 : reg ? reg : user.community,
+      enrolled_program:
+        zeroPro == 0 ? 0 : program ? program : user.enrolled_program,
+      current_term: zeroTerm == 0 ? 0 : term ? term : user.current_term,
+    }).catch((err) => {
+      console.log(err);
+      console.log(err.response);
+    });
+    if (result) {
+      navigation.navigate('Profile');
+    }
+  };
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== 'web') {
+        const status = await ImagePicker.requestCameraRollPermissionsAsync();
+        if (status !== 'granted') {
+          alert('We need your camera roll permissions to change your pictures');
+        }
+      }
+    })();
```

Comment:
i have it like this because it returns a promise and useEffect doesn’t expect the Promise returned in the callback. but yeah I'll define the function and call it to make it easier to read

Reply:
Okay makes sense, thanks!